test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+                     "trying to destroy chronicle on STORY_ACQUIRED state", tid);
+        }
+        else
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_ACQUIRED return value when "
+                     "trying to destroy chronicle on a state different than STORY_ACQUIRED", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_NOT_EXIST)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::STORY_DESTROYED ||
+           static_cast<ThreadState>(shared_state[tid]) == ThreadState::CHRONICLE_DESTROYED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                     "trying to destroy chronicle on a state STORY_DESTROYED or CHRONICLE_DESTROYED", tid);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_NOT_EXIST return value when "
+                      "trying to destroy chronicle on a state different than STORY_DESTROYED or CHRONICLE_DESTROYED"
+                      , tid);
+        }
+    }
+    else
+    {
+        LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- the return value is an unexpected one", tid);
+    }
+}
+
+void check_thread_finalized(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+    shared_state[tid] = static_cast<int>(ThreadState::THREAD_FINALIZED);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- State updated to: {}", tid, get_state_name(
+            ThreadState::THREAD_FINALIZED));
+}
+
+void handle_return_value(int tid, int ret, ThreadState success_state)
+{
+    switch(success_state)
+    {
+        case ThreadState::THREAD_INITIALIZED:
+            check_thread_initialization(tid, ret);
+            break;
+        case ThreadState::CHRONICLE_CREATED:
+            check_chronicle_created(tid, ret);
+            break;
+        case ThreadState::STORY_ACQUIRED:
+            check_story_acquired(tid, ret);
+            break;
+        case ThreadState::STORY_RELEASED:
+            check_story_released(tid, ret);
+            break;
+        case ThreadState::STORY_DESTROYED:
+            check_story_destroyed(tid, ret);
+            break;
+        case ThreadState::CHRONICLE_DESTROYED:
+            check_chronicle_destroyed(tid, ret);
+            break;
+        case ThreadState::THREAD_FINALIZED:
+            check_thread_finalized(tid, ret);
+            break;
+        case ThreadState::UNKNOWN:
+        default:
+            break;
+    }
+    //std::cout << "Thread " << tid << " checked its state: "
+    //          << get_state_name(static_cast<ThreadState>(shared_state[tid])) << std::endl;
+}
+
+void thread_body(struct thread_arg*t)
+{
+    // Thread Initialized
+    handle_return_value(t->tid, 0, ThreadState::THREAD_INITIALIZED);
+
```

Comment:
Do you think it's necessary to add random sleep to increase the possibility of different mixtures of states in related threads to increase the coverage of this test? For example, if there is no sleep like this, it is much less likely that a thread reaches `DestroyChronicle` while a related thread is still calling `CreateChronicle`.

Reply:
Like this a lot.

Reply:
I'm thinking that you can hold on on the randomized sleep() request. Let's wrap up this ticket with the logical/functional changes. Randomized sleep in the thread execution is a a good idea but I suggest we make it in the next round of changes together with the randomized chronicle/story to thread mapping.